### DISCUSSION TITLE
Is there a way to install eCAL python binding using pip? (arm64)

### DISCUSSION THREAD

**Discussion by chengguizi** [eclipse-ecal/ecal#839] [Q&A] [answered]
Currently the official way to install python binding is through system installer apt install python-ecal5.

However sometimes we need to install the eCAL package in a python venv, but seems that a pip package is not available yet.

I tried to find wheel file, and I found them here:
https://eclipse-ecal.github.io/ecal/_download_archive/download_archive_ecal_5_10_2.html 
https://github.com/eclipse-ecal/ecal/releases/tag/v5.10.2 

However, only x86_64 is available. How do I proceed if I want to use it on arm64? Thanks!

**Comment by KerstinKeller** [comment, marked as answer]
Hi, yes you are right. At the moment, no python pip package is available. It is on our list, but we have not yet tackled it. In order to publish the package on PyPi, you need to build a manylinux package which will work on all Unix flavors. This comes with a few challenges.
We do offer whls as outputs of the Github Action build agents we have on github actions, mainly Windows, Ubuntu 18 / 22 / 22 and Mac.
Our arm builds are done by canonical to create system packages but we have no runner and hence no .whl artifacts on Github Actions.
I am afraid you will have to (cross)compile them yourselves, if you need them.

**Reply by chengguizi** [reply]
No worries, thanks for the explanation! It is still just a nice to have currently. For now, I will rely on ubuntu system pacakage for easy installation on arm64 platforms.
